### PR TITLE
Refactor exceptions to share more code

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@chfast would you mind looking into the refactoring and giving hints how to fix it? :)

**Comment by jakelang** [comment]
@axic you must specify inherited constructors in the child class.
`using HeraException::HeraException` should do the trick

**Comment by jakelang** [comment]
fixed it @axic

File: src/exceptions.h [hunk 1]
```diff
@@ -28,16 +28,24 @@ namespace HeraVM {
 
 class HeraException : public std::exception {
 public:
-  explicit HeraException(std::string _msg): msg(std::move(_msg)) {}
```
Comment (axic) [inline src/exceptions.h]: Does this needs to be removed?

Comment (axic) [reply src/exceptions.h]: @jakelang @chfast ping ?

**Pull request merged**